lib/partitioner/go.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'cocaine'
+require 'set'
+require 'partitioner/dependency_analysis'
+
+module Partitioner
+  # This partitioner shard Go repos
+  class Go < DependencyAnalysis
+
+    def initialize(build, kochiku_yml)
+      super(build, kochiku_yml, {})
+    end
+
+    def partitioner_type
+      'go'
+    end
+
+    def all_modules
+      return @all_modules if @all_modules
+
+      @all_moduless = module_dependency_map.keys
+    end
+
+    def module_dependency_map
+      return @module_dependency_map if @module_dependency_map
+
+      @module_dependency_map = {}
+
+      # NOTE(kaneda): THIS MIGHT NOT WORK?
+      GitRepo.inside_copy(@build.repository, @build.ref) do
+        outputs = Cocaine::CommandLine.new("go list -f '{{.ImportPath}}:{{.Imports}}' ./...").run.split("\n")
```

Comment:
go1.9 changes the behavior of `go list` to omit vendored packages by default. I don't see a flag that would help re-include them. This will mean that the dependency map returned by this method won't be complete. I haven't made an attempt to learn the rest of the code well enough to evaluate the impact of this behavior.